src/storage/src/sink/kafka.rs

```diff
@@ -348,43 +416,58 @@ struct ProgressInitState {
 }
 
 impl ProgressInitState {
-    fn to_running(self, gate_ts: Rc<Cell<Option<Timestamp>>>) -> ProgressRunningState {
+    fn to_running(
+        self,
+        gate_ts: Rc<Cell<Option<Timestamp>>>,
+        healthchecker: Arc<Mutex<Option<Healthchecker>>>,
+    ) -> ProgressRunningState {
         ProgressRunningState {
             topic: self.topic,
             key: self.key,
             gate_ts,
+            healthchecker,
         }
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 struct ProgressRunningState {
     topic: String,
     key: String,
     gate_ts: Rc<Cell<Option<Timestamp>>>,
+    // Using Arc<Mutex> because clippy complains about holding a RefCell reference across await points if we use
+    // an `Rc<RefCell>`.
+    healthchecker: Arc<Mutex<Option<Healthchecker>>>,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 enum KafkaSinkStateEnum {
     // Initialize ourselves as a transactional producer with Kafka
     // Note that this only runs once across all workers - it should only execute
     // for the worker that will actually be publishing to kafka
-    Init(Option<ProgressInitState>),
```

Comment:
this is unrelated cleanup that we can do now that we are always in "reuse topic" mode!